### PR TITLE
Improved insights history

### PR DESCRIPTION
**Pull request by timgl**
## Changes

Adds a "History" tab to insights, with pretty blocks of stuff you've recently done or saved. This is ready for an initial look/some feedback. There are still a few bugs I need to solve.

image

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by Twixes** [comment]
This shouldn't really be an Insight type as it now is in this PR, that's somewhat confusing. It's kind of a metainsight, so I'd argue it should still be a button on the right:

Screen Shot 2020-12-14 at 13 44 09

File: frontend/src/scenes/retention/retentionTableLogic.ts [hunk 1]
```diff
@@ -73,26 +68,34 @@ function toUrlParams(values: Record<string, unknown>, extraVals?: Record<string,
...
     },
-    loaders: ({ values }) => ({
-        retention: {
-            __default: ({} as Record<string, unknown>) || Array,
-            loadRetention: async (_: any, breakpoint) => {
+    loaders: ({ values, props }) => ({
+        results: {
+            __default: [],
+            loadResults: async (refresh = false, breakpoint) => {
+                console.log(props.cachedResults, refresh)
```
Comment (EDsCODE) [inline frontend/src/scenes/retention/retentionTableLogic.ts:91]: printer

**Review by EDsCODE — COMMENTED** [review]
initial scan. going to QA now

File: posthog/tasks/update_cache.py [hunk 2]
```diff
@@ -23,7 +24,7 @@ def update_cache_item(key: str, cache_type: str, payload: dict) -> None:
 
     result: Optional[Union[List, Dict]] = None
     filter_dict = json.loads(payload["filter"])
-    filter = Filter(data=filter_dict)
+    filter = get_filter(data=filter_dict, team=Team(pk=payload["team_id"]))
     if cache_type == CacheType.TRENDS:
         result = _calculate_trends(filter, key, int(payload["team_id"]))
     elif cache_type == CacheType.FUNNEL:
```
[the remark this answers is not in the document]
Comment (timgl) [reply posthog/tasks/update_cache.py:30]: Yeah good point, I sort of wanted to push this as the PR is already getting very big, and the updating of cached retention/sessions etc doesn't work right now anyway. Happy to grab it now or do it in a new PR

**Review by EDsCODE — COMMENTED** [review]
Very cool, few comments and style considerations. I'm wondering if it might be better to just make history a subtab under the insights tab. Now that it's not a sidebar and a whole page for itself, it seems like it could exist on its own. 

1. Weird empty tab you can click on
Screen Shot 2020-12-15 at 1 48 30 PM

2. When clicking history, it might be better to default to recent first, especially if there are no saved ones yet
3. I think the history button should be more distinct and not a tab. Since it's not grouped with other tabs it looks like floating text
4. When you hover over a history dashboard if gives you the pointer symbol but the graph itself isn't clickable, only the title.
5. retention graphs don't have a visual/thumbnail in the history panels right away (it shows up after going away from the page and back)
6. For recent history items, I think it would make more sense to not allow people to "move" them. Right now you can move a history item to a dashboard and it will be added to the dashboard with the name "unsaved query"

**Comment by timgl** [comment]
Thanks, ready for another look if you want. Fixed all of your points. I've left the history button on the top right, as it'll work better with the new nav bar.

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**